### PR TITLE
[Bug] + [Tests] Fix for SAPDownloadedPackage Signature Verification

### PR DESCRIPTION
**Pull request by SebastianWild**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [ ] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [ ] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (if applicable)
* [ ] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

## Description

- Fix issue where a missing public key or failed signature creation of a
signature data inside `SAP_TEKSignature` would fail the entire process,
even if a valid signature would have been later in the process. This is not an issue with the current landscapes but is an edge case that should be handled.
- Added unit tests for `SAPDownloadedPackage`

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift [hunk 1]
```diff
@@ -85,7 +85,7 @@ final class HTTPClient: Client {
 				}
 				do {
 					// Configuration File Signature must be checked by the application since it is not verified by the operating system
-					guard try package.verifySignature() else {
+					guard package.verifySignature() else {
 						logError(message: "Failed to verify configuration data signature")
 						completion(nil)
```
Comment (haosap) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift:90]: shall we notify the callee that the specific error message?

**Review by haosap — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift (hunk 1 above)
Comment (ChristianKienle) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift:90]: he did not change that line

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/SAPDownloadedPackage.swift [hunk 2]
```diff
@@ -52,16 +52,21 @@ extension SAPDownloadedPackage {
...
 		for signatureEntry in parsedSignatureFile.signatures {
 			let signatureData: Data = signatureEntry.signature
-			let publicKey = try keystore.publicKey(for: signatureEntry.signatureInfo.appBundleID)
-			let signature = try P256.Signing.ECDSASignature(derRepresentation: signatureData)
+			guard
+				let publicKey = try? keystore.publicKey(for: signatureEntry.signatureInfo.appBundleID),
+				let signature = try? P256.Signing.ECDSASignature(derRepresentation: signatureData)
+			else {
+				logError(message: "Could not validate signature of downloaded package", level: .warning)
+				continue
```
[the remark this answers is not in the document]
Comment (ChristianKienle) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/SAPDownloadedPackage.swift:68]: I think we should review the verification code once again. I think it is not correct to be honest...

[the remark this answers is not in the document]
Comment (haosap) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/SAPDownloadedPackage.swift:68]: Ok. Let's invite @SebastianWild to have a meeting this evening.

[the remark this answers is not in the document]
Comment (ChristopherSchmitz) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/SAPDownloadedPackage.swift:68]: @ChristianKienle  could you please be more precise why you see the code as not correct. To be precise:
- the backend is leading
- if multiple signatures are returned:
--> the first matching signature returns the verify function
--> an error in previous runs signature checks will not break the loop and allow subsequent entries to apply for a correct signature

What is missing from your side here?

Please note: Google and Apple announced to retire .signatureInfo.appBundleID. As a result we move to the bundle ID of the signed application

**Review by ChristianKienle — APPROVED** [review]

**Review by haosap — APPROVED** [review]

**Review by ChristianKienle — APPROVED** [review]
